src/caldavclient.cpp

```diff
@@ -89,13 +95,32 @@ bool CalDavClient::init()
 
     mNAManager = new QNetworkAccessManager(this);
 
-    if (initConfig()) {
-        return true;
-    } else {
-        // Uninitialize everything that was initialized before failure.
-        uninit();
+    mManager = new Accounts::Manager(this);
+
+    Accounts::Service srv;
+    Accounts::Account *account = getAccountForCalendars(&srv);
+    if (!account) {
+        return false;
+    }
+
+    mSettings.setServerAddress(account->value("server_address").toString());
+    if (mSettings.serverAddress().isEmpty()) {
+        qCWarning(lcCalDav) << "remote_address not found in service settings";
+        return false;
+    }
+    mSettings.setDavRootPath(account->value("webdav_path").toString());
+    mSettings.setIgnoreSSLErrors(account->value("ignore_ssl_errors").toBool());
+    account->selectService(Accounts::Service());
+
+    mAuth = new AuthHandler(mManager, mAccountId, srv.name());
```

Comment:
Existing thing, but is this now leaking memory? Should there be e.g. parent argument passed with 'this'?

Reply:
Good catch. I've added a `this` argument.